testutils/src/main/java/com/amplifyframework/testutils/ObjectValidatorUtils.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.testutils;
+
+import org.junit.Assert;
+
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Utility class that can be used to test core
+ * Object behavior of a class that implements its own
+ * equals, hashCode and toString methods.
+ */
+public final class ObjectValidatorUtils {
+
+    /**
+     * Hidding default constructor of utility class.
+     */
+    private ObjectValidatorUtils() {}
+
+    /**
+     * Function to make some basic assertions as it related to the
+     * proper implementation of equals, hashCode and toString.
+     * @param subject1 An instance of T
+     * @param suject2 An instance of T that is not equal to subject1.
+     * @param subject3ThatEqualsSubject1 An instance of T such that when passed as
+     *                                   a parameter to subject1.equals, will return true.
+     * @param <T> The type being tested.
+     */
+    public static <T> void assertCoreObjectBehavior(T subject1, T suject2, T subject3ThatEqualsSubject1) {
```

Comment:
Assuming this is contained `EqualsToStringHashValidator`, then maybe:

```suggestion
    public static <T> void validate(T reference, T equalsReference, T notEqualsReference) {
```

Reply:
Refactored this logic into its own class in testutils.